bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/ComponentSwitch.java

```diff
@@ -37,11 +37,11 @@
         protected boolean optimistic = false;
 
         protected String state_topic = "";
-        protected String state_on = "true";
-        protected String state_off = "false";
+        protected String state_on = "ON";
+        protected String state_off = "OFF";
```

Comment:
The states are optional ans should be equal to the payloads, if not present.
So they should be `@Nullable` and then:
```
        String sOn = channelConfiguration.state_on;
        if (sOn == null) {
            sOn = channelConfiguration.payload_on;
        }
        String sOff = channelConfiguration.state_off;
        if (sOff == null) {
            sOff = channelConfiguration.payload_off;
        }

        OnOffValue value = new OnOffValue(sOn, sOff, channelConfiguration.payload_on, channelConfiguration.payload_off);

```

Reply:
Good point

Reply:
I do not get youe comment (the edit part)
The config is deserialized in the parent class from a JSON, which is read from an MQTT topic.
So the configuration is in use.
But we can for sure handle this in another PR